packages/features/pbac/services/permission-check.service.ts

```diff
@@ -0,0 +1,59 @@
+import { MembershipRole } from "@calcom/prisma/enums";
+
+import type { PermissionString, Resource, Action } from "../types/permission-registry";
+import type { RoleService } from "./role.service";
+
+export class PermissionCheckService {
+  constructor(private roleService: RoleService) {}
+
+  async hasPermission(
+    membership: { role: MembershipRole; customRoleId: string | null },
+    permission: PermissionString
+  ): Promise<boolean> {
+    // First check if user has a custom role
+    if (membership.customRoleId) {
+      const hasCustomPermission = await this.checkCustomRolePermission(membership.customRoleId, permission);
+      if (hasCustomPermission) return true;
+    }
+
+    // Fall back to default role permissions
+    return this.checkDefaultRolePermission(membership.role, permission);
+  }
+
+  private checkDefaultRolePermission(role: MembershipRole, permission: PermissionString): boolean {
+    // Define default permissions for each role
+    const defaultPermissions: Record<MembershipRole, PermissionString[]> = {
+      [MembershipRole.OWNER]: ["*.*"], // Owner has all permissions
+      [MembershipRole.ADMIN]: [
+        "booking.*",
+        "eventType.*",
+        "team.invite",
+        "team.remove",
+        "organization.manage",
+      ],
+      [MembershipRole.MEMBER]: ["booking.read", "eventType.read", "team.read"],
+    };
```

Comment:
We need to come up with this list of permissions for each team role that we currently have. Gonna be pretty dificult to do i think 🤔